torch/_inductor/select_algorithm.py

```diff
@@ -366,6 +372,46 @@ def __init__(
         self.template_out: Optional[str] = None
         self.ops_handler: Optional[V.WrapperHandler] = None  # type: ignore[name-defined]
 
+        # The generated code is not dependent on the input nodes names, or symbolic sizes names.
+        # However, some of the variables returned by generate_and_load that are computed during the
+        # triton template expansions (code generation) are dependent on those.
+        # In order to cache the code generation and avoid redoing it for similar inputs that varies only by
+        # input names or symbol names, we do a record and replay method.
+        # During template expansions we record all function calls that change input_dependent_preserved_state
+        # and replay them on a cache hit to regenerate them.  Namely we want to be able to regenerate:
+        # 1) self.args.input_buffers
+        # 2) self.args.sizevars
+        # 3) self.prologue_supported_inputs
+
+        self.cached_replay_events: RecordedEventsType = []
+
+    def input_dependent_preserved_state(self) -> str:
+        return repr(
+            [
+                self.args.input_buffers,
+                self.args.sizevars,
```

Comment:
nit: can we include this list above, in __init__, so it will be more visible to people who add more attrs ?

Reply:
how can we do that? like in the comment?